### PR TITLE
Issue 87: Improve policies - Part 1: outputs

### PR DESCRIPTION
**Pull request by LourensVeen**
This is the first part of the implementation of policies V2. It adds an `output` attribute to `ResultOfIn` rules, and modifies the evaluation rules so that the rules apply to the specified output. Handy for steps that produce some sensitive and some non-sensitive outputs.

Partially implements #87.

### CODE REVIEW

File: mahiru/policy/rules.py [hunk 1]
```diff
@@ -122,13 +122,15 @@ def __init__(
...
             compute_asset: Union[str, Identifier],
+            output: str,
             collection: Union[str, Identifier]
             ) -> None:
         """Create a ResultOfIn rule.
 
         Args:
             data_asset: The source data asset.
             compute_asset: The compute asset used to process the data.
+            output: The workflow output that produced the result Asset.
```
Comment (svenvanderburg) [inline mahiru/policy/rules.py]: So what you mean with 'output' refers to sort of the variable name of the workflow output value that is then feed into the result Asset? It's a bit implicit from the docstring alone, this is what I figured from the examples. Maybe you can add a bit more details here or add it to your 'glossary' overview.

**Review by svenvanderburg — APPROVED** [review]
Looks good. I just added one comment for clarification.

File: mahiru/policy/rules.py (hunk 1 above)
Comment (LourensVeen) [reply mahiru/policy/rules.py]: Yes, it is the name of the output of the workflow step that produced the asset that's going to be in the collection. I've tried to clarify that a bit, how's this?

Comment (svenvanderburg) [reply mahiru/policy/rules.py]: Yes, looks good! :)

**Pull request merged**